### PR TITLE
Maintain Topic._messages ordering by seq.

### PR DESCRIPTION
**Pull request by aforge**
This invariant breaks in a few cases:
- when updating deletion gaps
- when Message.seq is updated with the server specified value.

### CODE REVIEW

File: src/tinode.js [hunk 1]
```diff
@@ -3543,7 +3543,21 @@ Topic.prototype = {
...
+      pub.seq = newSeq;
+      if (0 <= idx && idx < numMessages) {
+        // this._messages are sorted by `seq`.
+        // If changing pub.seq to newSeq breaks the invariant, fix it.
+        if ((idx > 0 && this._messages.getAt(idx - 1).seq >= newSeq) ||
+          (idx + 1 < numMessages && newSeq < this._messages.getAt(idx + 1).seq <= newSeq)) {
+          this._messages.delAt(idx);
+          this._messages.put(pub);
+        }
+      }
```
Comment (or-else) [inline src/tinode.js]: Loolks like you are editing some older version of tinode-js (master?).  This has been fixed a while ago:
https://github.com/tinode/tinode-js/commit/7b84cce6d3b48e08e1e157cab55551216773242c

File: src/tinode.js [hunk 2]
```diff
@@ -4117,7 +4131,7 @@ Topic.prototype = {
     if (cb) {
       let startIdx = typeof sinceId == 'number' ? this._messages.find({
         seq: sinceId
-      }) : undefined;
+      }, true) : undefined;
```
Comment (or-else) [inline src/tinode.js]: Good catch. Thanks!

File: src/tinode.js [hunk 3]
```diff
@@ -4716,6 +4730,10 @@ Topic.prototype = {
     // as placeholers for deleted ranges.
     // The messages are iterated by seq ID in ascending order.
     this._messages.forEach((data) => {
+      // Message not sent.
```
Comment (or-else) [inline src/tinode.js]: Maybe expand the comment '// Do not create a gap between the last sent message and the first unsent.'

File: src/tinode.js (hunk 1 above)
Comment (aforge) [reply src/tinode.js]: Moved this code to swapMessage().

File: src/tinode.js (hunk 3 above)
Comment (aforge) [reply src/tinode.js]: Done.

File: src/tinode.js [hunk 4]
```diff
@@ -4231,12 +4217,36 @@ Topic.prototype = {
...
   },
 
+  /**
+   * Update message's seqId.
+   * @memberof Tinode.Topic#
+   *
+   * @param {Object} pub message object.
+   * @param {integer} newSeqId new seq id for pub.
+   */
+  swapMessage: function(pub, newSeqId) {
```
Comment (or-else) [inline src/tinode.js]: swapMessageId?

Comment (aforge) [reply src/tinode.js]: Done.

**Pull request merged**